src/Octoshift/AwsCredentialProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using Amazon.Runtime;
+
+namespace OctoshiftCLI;
+
+public class AwsCredentialProvider
+{
+    private readonly AWSArgs _awsArgs;
+    public AwsCredentialProvider(AWSArgs awsArgs)
+    {
+        _awsArgs = awsArgs;
+    }
+
+    public AWSCredentials GetCredentials()
+    {
+        if (string.IsNullOrEmpty(_awsArgs.AwsCredentialType)) //load credentials from provider chain
+        {
+            AWSCredentials result = SessionCredentials();
```

Comment:
I am guessing you would use another tool before you call `gh gei` to generate the session credentials, and then you would pass them to the `gh gei` CLI? How would you normally generate the session credentials in practice?

Reply:
When I do some tests on my local laptop, I can use `aws cli` to get session credentials, such as `aws sts get-session-token`. And then I can store credentials as ENV variables and call `gh gei` CLI.